### ISSUE TITLE
Instability and Unpredictable Snapshot Order in Tests using Insta and SQLx

### ISSUE THREAD

**Issue by flappyBug** [mitsuhiko/insta#434]
### What happened?

I'm experiencing instability and unpredictable behavior when running tests with Insta and SQLx. Specifically, the order of snapshots seems to be inconsistent, leading to test failures.

### Reproduction steps

Reproducible Code:

```rust
#[cfg(test)]
mod tests {
    use sqlx::{Sqlite, Pool, Row};

    #[sqlx::test]
    fn snapshot1(pool: Pool<Sqlite>) -> sqlx::Result<()> {
        let foo = sqlx::query("SELECT 1 as 'bar'")
            .fetch_one(&pool)
            .await?;

        insta::assert_debug_snapshot!(foo.get::<i32, _>("bar"));

        Ok(())
    }

    #[sqlx::test]
    fn snapshot2(pool: Pool<Sqlite>) -> sqlx::Result<()> {
        let foo = sqlx::query("SELECT 2 as 'bar'")
            .fetch_one(&pool)
            .await?;

        insta::assert_debug_snapshot!(foo.get::<i32, _>("bar"));

        Ok(())
    }
}
```
My dependencies:
```toml
[dependencies]
sqlx = { version = "0.7.3", default-features = false, features = [
    "sqlite",
    "runtime-tokio",
    "macros",
    "migrate",
    "chrono",
    "uuid",
] }
[dev-dependencies]
insta = {version = "1.34.0", features = ["redactions", "filters"]}

[profile.dev.package.sqlx-macros]
opt-level = 3

[profile.dev.package]
insta.opt-level = 3
similar.opt-level = 3
```

Then run `cargo insta test --review`. Accept once, and run multiple times. Sometimes tests fail due to unexpected snapshot order.

My OS and chip:
`macOS 14.1.1 (23B81), Apple M2 Pro`

### Insta Version

1.34.0

### rustc Version

rustc 1.73.0 (cc66ad468 2023-10-03)

### What did you expect?

I expect the snapshots to maintain a consistent order during test execution, ensuring reliable and stable results.

**Comment by max-sixty** [comment]
This might be obvious to others, but why does it matter if they're run in a different order? Is the result back from the queries different depending on the order (I'm guessing not)?

**Comment by flappyBug** [comment]
@max-sixty  My provided code generates two snapshots. For example, 
- crate__tests__inner.snap:
```
---
source: src/lib.rs
expression: "foo.get::<i32, _>(\"bar\")"
---
1
```
- crate__tests__inner-2.snap:
```
crate__test__inner.snap`
---
source: src/lib.rs
expression: "foo.get::<i32, _>(\"bar\")"
---
2
```

But it have chance to mess up the order. The first test is trying to match the second snap, and the second test is trying to match the first snap.

**Comment by max-sixty** [comment]
Generally the default test comes from the function name, so these would be named differently. Is the annotation somehow not making that possible? Does it work without the annotation?

**Comment by flappyBug** [comment]
@max-sixty The annotation introduced an async runtime, and a sqlite pool instance. I cannot figure out how to expand the test with cargo expand. But I dug into the code, it's something like:
```rust
    #[::core::prelude::v1::test]
    fn snapshot2() -> sqlx::Result<()> {
        async fn inner(pool: Pool<Sqlite>) -> sqlx::Result<()> {
            let foo = sqlx::query("SELECT 2 as 'bar'").fetch_one(&pool).await?;

            insta::assert_debug_snapshot!(foo.get::<i32, _>("bar"));

            Ok(())
        }

        let args = ::sqlx::testing::TestArgs::new(concat!(module_path!(), "::", stringify!(#name)));

        // #migrations

        // args.fixtures(&[#(#fixtures),*]);

        // We need to give a coercion site or else we get "unimplemented trait" errors.
        let f: fn(Pool<Sqlite>) -> _ = inner;

        ::sqlx::testing::TestFn::run_test(f, args)
    }
```

Ahh yes, the annotation introduced an `inner` function.

**Comment by flappyBug** [comment]
I made an even more simple code to reproduce it.
```
#[cfg(test)]
mod tests {
    fn inner(i: i32) {
        insta::assert_debug_snapshot!(i);
    }

    #[test]
    fn snapshot1() {
        inner(1);
    }

    #[test]
    fn snapshot2() {
        inner(2);
    }
}
```

**Comment by max-sixty** [comment]
You could assign names to the snapshots yourself? Or use inline snapshots?

Otherwise I'm not sure there's much insta can do (though not a maintainer and others may have ideas)

**Comment by flappyBug** [comment]
Yes, I just get to know that I can assign name explicitly. I'm wondering if we can use the name of outer most function in a module as the name used as snapshot suffix. Or we can use the exact function annotated with `#[test]`. 

I'm pretty sure at least the first approach is feasible. Any opinion?

**Comment by ciffelia** [comment]
Fixed in sqlx 0.7.4.
https://github.com/launchbadge/sqlx/blob/v0.7.4/CHANGELOG.md
